src/generator/type_inference_visitor.ts

```diff
@@ -132,7 +132,19 @@ class TypeInferenceVisitor extends Visitor {
 
         if (node.computed) {
             super.visit(node.property);
-            type = WebAssemblyType.INT_32;
+
+            const arrayType = this.getTypeOfExpression(node.object);
+
+            switch (arrayType) {
+                case WebAssemblyType.INT_32_ARRAY:
+                    type = WebAssemblyType.INT_32;
+                    break;
+                case WebAssemblyType.FLOAT_64_ARRAY:
+                    type = WebAssemblyType.FLOAT_64;
+                    break;
+                default:
+                    throw new Error(`Unknown array type ${arrayType}`);
```

Comment:
This only prints a number as the type in the error message. Use `WebAssemblyType[arrayType]` instead.